### DISCUSSION TITLE
추상 클래스는 무쓸모인가요??

### DISCUSSION THREAD

**Discussion by yxxnghwan** [woowa-Effective-Java-Study/effective-java-study#12] [General]
인터페이스에 디폴트 메서드가 생기면서 인터페이스도 특정 메서드에 구현을 제공해줄 수 있는데, 추상 클래스는 이제 설 자리가 없나요?

**Comment by yxxnghwan** [comment]
디폴트 메서드는 골격 구현을 위한게 아니라 유지보수를 위한 거라서 골격 구현 목적으로 쓰면 안된다.
API가 자바 버전이 올라가면서 인터페이스에 새 메서드를 추가할 때 이미 그 인터페이스를 구현하던 클래스들은 새 메서드를 다시 구현해야 하기 때문에,
그거를 편하게 하고자 디폴트 메서드가 나와서 구현 클래스에서 새 메서드를 구현하지 않고 추가되는 버전에서 넣어주려고 만든거.
골격구현 할때는 추상 클래스를 쓰는게 좋다.

**Reply by OzRagwort** [reply]
키워드 : 스켈레톤 클래스

목적이 좀 다른것 같다.

디폴트 메서드는 레거시를 위해 만들어 졌는데 기존 인터페이스에 뭔가가 추가가 되면 그 인터페이스를 쓰는 모든애들에 override를 해줘야 하니까 힘들다. 그래서 디폴트 메서드로 만들면 그렇게 안해도 되니까 좋다!

스켈레톤 클래스 : 추상 클래스, 인터페이스의 장점을 섞고 싶을때? ㅇ? 골격구현? 

모르겠다 ㅇ>-<
